File: 70. Climbing Stairs.js
// https://leetcode.com/problems/climbing-stairs/

var climbStairs = function (n) {
    var oneBack = 2;
    var twoBack = 1;

    var curr;

    if (n <= 2) return n;

    for (var i = 3; i <= n; i++) {
        curr = oneBack + twoBack;
        twoBack = oneBack;
        oneBack = curr;
    }

    return curr;
};

console.log(climbStairs(5))

File: Dynamic Programming/70_Climbing_Stairs.kt
// 70. Climbing Stairs 

fun main(){
    println(climbStairs(5))
}

fun climbStairs(n: Int): Int {
    var oneStepBack = 2
    var twoStepsBack = 1
    
    if(n <= 2) return n
    
    for(i in 3..n) {
        var temp = oneStepBack
        oneStepBack = oneStepBack + twoStepsBack
        twoStepsBack = temp
    }
    
    return oneStepBack
}